models/articles.models.js

```diff
@@ -25,12 +23,14 @@ exports.fetchArticleById = (article_id) => {
         })
 }
 
-exports.checkArticleExists = (article_id) => {
+exports.updateArticleVotesById = (article_id, inc_votes) => {
     return db
-    .query("SELECT * FROM articles WHERE article_id = $1", [article_id])
+    .query("SELECT votes FROM articles WHERE article_id = $1", [article_id])
     .then(({rows}) => {
-        if(!rows.length){
-            return Promise.reject({status: 404, msg: "Article Not Found"})
-        }
+        const oldVotes = rows[0];
+        const newVotes = oldVotes.votes + inc_votes;
+        return db
+            .query("UPDATE articles SET votes = $1 WHERE article_id = $2 RETURNING *", [newVotes, article_id])
+            .then(({rows}) => rows[0])
```

Comment:
Instead of using two queries, one to find the votes and another to increase them, you can update votes in one step with `votes = votes + $1`

Reply:
oh that is amazing!! I have refactored this, thanks so much for the feedback 🙌